### PR TITLE
Plumbing application for probable AuthN/Z changes

### PR DESCRIPTION
**Pull request by jeremyf**
Plumbing application for probable AuthN/Z changes

For further details, core members can read:

- [Forem Article Authorization Discrepencies][1]
- [Forem Authentication and Authorization Principles][2]

(Note: there's nothing "private" about these articles, I happened to
post them on https://forem.team for internal discussion.)

Related to #16529

[1]:https://forem.team/jeremy/forem-article-authorization-discrepencies-45df
[2]:https://forem.team/jeremy/forem-authentication-and-authorization-principles-58l3

### CODE REVIEW

**Review by citizen428 — APPROVED** [review]

File: config/locales/misc/en.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ en:
     application_policy:
       must_be_logged_in: must be logged in
       you_must_be_logged_in: You must be logged in
+      you_have_a_suspended_account: You have a suspended account
```
Comment (aitchiss) [inline config/locales/misc/en.yml:34]: I feel like "Your account is suspended" would be less ambiguous (it might just be me, and feel free to ignore)

**Review by aitchiss — COMMENTED** [review]
The overall approach makes sense to me & I appreciate all the comments noting what's deprecated / what's to come 👍 

Although I can follow along, I feel like the second approval ought to be someone more familiar with the ruby though

File: config/locales/misc/en.yml (hunk 1 above)
Comment (jeremyf) [reply config/locales/misc/en.yml:34]: I like this wording and will be adding it.

File: config/locales/misc/en.yml [hunk 2]
```diff
@@ -31,6 +31,7 @@ en:
     application_policy:
       must_be_logged_in: must be logged in
       you_must_be_logged_in: You must be logged in
+      your_account_is_suspended: Your account is suspended
```
Comment (jeremyf) [inline config/locales/misc/en.yml:34]: Reading note, this is not yet used as I have a follow-up PR that will use this.  But it makes some sense to include.

File: config/locales/misc/fr.yml [hunk 3]
```diff
@@ -31,6 +31,7 @@ fr:
     application_policy:
       must_be_logged_in: must be logged in
       you_must_be_logged_in: You must be logged in
+      your_account_is_suspended: Your account is suspended
```
Comment (jeremyf) [inline config/locales/misc/fr.yml:34]: Reading note, this is not yet used as I have a follow-up PR that will use this.  But it makes some sense to include.

**Review by dscottS3 — APPROVED** [review]
LGTM! Appreciate the push toward consistency

**Pull request merged**